internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -983,17 +982,8 @@ func (s *Server) processPullRequestEvent(
 	}
 
 	ghRepo := event.GetRepo()
-	repoEnt, err := s.fetchRepo(ctx, &repo{
-		ID:      ghRepo.ID,
-		HTMLURL: ghRepo.HTMLURL,
-		Private: ghRepo.Private,
-	})
-	if err != nil {
-		return nil, err
-	}
-
 	pullProps, err := properties.NewProperties(map[string]any{
-		properties.PropertyUpstreamID: event.GetPullRequest().GetID(),
+		properties.PropertyUpstreamID: strconv.FormatInt(event.GetPullRequest().GetID(), 10),
```

Comment:
nit: you probably want to create a function to format the ID.

Reply:
I added a function using generics and Go's `x/constraints` module. This might also be useful for the similar functions in the GL provider.

This also uncovered a bug in the original properties, luckily the handlers refresh the PRs so we'd paper over the bug now, plus we also pass the name if we can't match the ID.